### ISSUE TITLE
questions about magic-numbers in dataset splitting

### ISSUE THREAD

**Issue by initzhang** [jadadong/GNS#2]
I have several questions about the statistics related to sampling and dataset splitting:
 
1. Could you please explain why you choose the specific splitting ratio for Train/Val/Test on the datasets?
<img width="834" alt="1" src="https://user-images.githubusercontent.com/33781006/137621200-fecfc161-f35c-42ea-8a8c-cb7e8c9ff43f.png">

2. I found that the #inputs statistics of OGBN-Papers100M will differ a lot if you use a larger training set. For example on the full graph, NS with 15-10-5 fanout only generates ~117406 inputs per mini-batch(because this dataset is quite dense). Do you have corresponding #cached nodes stats of OGBN-Papers100M when using a larger training set? 
<img width="503" alt="image" src="https://user-images.githubusercontent.com/33781006/137621245-9d735ff0-6710-4027-aae5-7ade3a80d916.png">

3. In the chapter 4.1 of the paper, you said that "The sampling fan-outs of each layer are 15, 10 for the third and second layer. We sample nodes in the first layer (input layer) only from the cache". Could you please elaborate how you set the fan-out for the first layer, e.g. is there a upper/lower bound?

Thank you for your time!

**Comment by initzhang** [comment]
Hi @zheng-da , could you please look at this issue? It seems that the repo owner is not active on GitHub, Thank you!

**Comment by zheng-da** [comment]
I think the data splits come from the original dataset. We didn't define our own data split.

2. we didn't try any other splits. as far as i remember, papers100M has a small number of nodes with labels. how do you increase the training set?

3. i believe we set the upper bound to be 15, 10.

**Comment by initzhang** [comment]
> I think the data splits come from the original dataset. We didn't define our own data split.
> 
> 1. we didn't try any other splits. as far as i remember, papers100M has a small number of nodes with labels. how do you increase the training set?
> 2. i believe we set the upper bound to be 15, 10.

Thank you for your reply!
The first two questions are my mistakes, sorry for that. But for the third question I am still confused. To be more specific, the fan-out of the baseline, a three-layer-GNN, is [15, 10, 5], while I cannot figure out the `X` for fan-out [15, 10, X] of GNS. Do we just accumulate all the first-hop neighbors of the target nodes in Global Cache?
